phoenix-core/src/main/java/org/apache/phoenix/compile/DeclareCursorCompiler.java

```diff
@@ -45,10 +45,11 @@
     private final Operation operation;
     private QueryPlan queryPlan;
 
-    public DeclareCursorCompiler(PhoenixStatement statement, Operation operation, QueryPlan queryPlan) {
+    public DeclareCursorCompiler(PhoenixStatement statement, Operation operation, QueryPlan queryPlan) throws SQLException {
         this.statement = statement;
         this.operation = operation;
-        this.queryPlan = queryPlan;
+        this.queryPlan = statement.getConnection().getQueryServices().getOptimizer()
```

Comment:
How does this change help?
By pre-optimizing the query, we force the next optimization plan to be a No-Op ?

Reply:
With this change we optimize the plan inside the CursorFetchPlan.
Later when the next optimize is called, the original CursorFetchPlan will be selected as there won't be any better plans.

Reply:
Please add a code comment with this information and a reference to this ticket.